kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerProducer.java

```diff
@@ -94,10 +95,13 @@ private void sendEvent(ProcessInstance processInstance,
     private void logError(Object value, Exception e) {
         logger.error("Error publishing event {}", value, e);
     }
-
+    
+    
     public void activate(DeploymentEvent event) {
-        ((InternalRegisterableItemsFactory) ((InternalRuntimeManager) event.getDeployedUnit().getRuntimeManager())
-                .getEnvironment().getRegisterableItemsFactory()).addProcessListener(this);
+        SimpleRegisterableItemsFactory itemsFactory = (SimpleRegisterableItemsFactory) ((InternalRuntimeManager) event
+                .getDeployedUnit().getRuntimeManager()).getEnvironment().getRegisterableItemsFactory();
+        itemsFactory.addWorkItemHandler("Send Task", SendTaskHandler.class);
```

Comment:
The problem with this approach is that using the default _SendTaskHandler_ class, only String messages could be sent:
https://github.com/kiegroup/jbpm/blob/62f92c98f5d3f68f08fd519638259be97aa2d574/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/handler/SendTaskHandler.java#L30

Then, this kind of tests with a custom message would fail:
https://github.com/kiegroup/droolsjbpm-integration/blob/dd06377f866ba814b45d9d654c64c3f754084622/kie-spring-boot/kie-spring-boot-samples/kie-server-spring-boot-sample/src/test/java/org/kie/server/springboot/samples/KafkaProducerHappyPathTest.java#L195

wdyt?

Reply:
I will change the default handler to not assume string